components/core/src/metrics.rs

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) 2016 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::thread;
+use std::sync::mpsc::{channel, sync_channel, Sender, Receiver, SyncSender};
+use std::sync::{Once, ONCE_INIT};
+use statsd::Client;
+use env;
+
+// Statsd Application name
+pub const APP_NAME: &'static str = "bldr";
```

Comment:
Why is this bldr and not `hab`? I assume its because we're going to be embedding this in various bldr components, but what about the `hap-sup` itself, which also could use a generic metrics endpoint, both for sending and recieving?